### PR TITLE
task(payments): Update strings referencing 'Firefox accounts'

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We want to start the localization effort of updating strings from Firefox to Mozilla

## This pull request

- Creates new l10n strings / ids using the Mozilla branding

## Issue that this pull request solves

Closes: FXA-7991

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-payments-server/src/components/NewUserEmailForm/en.ftl [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## Component - NewUserEmailForm
 
 new-user-sign-in-link = Already have a { -brand-name-firefox } account? <a>Sign in</a>
+new-user-sign-in-link-2 = Already have a { -brand-name-mozilla } account? <a>Sign in</a>
```
Comment (bcolsson) [inline packages/fxa-payments-server/src/components/NewUserEmailForm/en.ftl:4]:
```suggestion
new-user-sign-in-link-2 = Already have a { -product-mozilla-account }? <a>Sign in</a>
```
Best to update this string to use the correct term here.

File: packages/fxa-payments-server/src/routes/Checkout/en.ftl [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ## Routes - Checkout - New user
 
 new-user-step-1 = 1. Create a { -brand-name-firefox } account
+new-user-step-1-2 = 1. Create a { -brand-name-mozilla } account
```
Comment (bcolsson) [inline packages/fxa-payments-server/src/routes/Checkout/en.ftl:4]:
```suggestion
new-user-step-1-2 = 1. Create a { -product-mozilla-account }
```

**Review by bcolsson — CHANGES_REQUESTED** [review]

**Review by sardesam — APPROVED** [review]
r+ from me but I do have a question.

`-product-mozilla-account` and `-product-mozilla-account` (edit: `-product-firefox-account`) are both declared in `packages/fxa-shared/l10n/branding.ftl` with the former replacing `-brand-name-firefox` in `NewUserEmailForm` and `routes/Checkout`. However, `packages/fxa-shared/l10n/branding.ftl` has `-brand-mozilla` and `-brand-firefox`, yet we're using `-brand-name-mozilla` and `-brand-name-firefox` defined in `packages/fxa-payments-server/src/branding.ftl` instead.

I'm curious why it's structured this way. Should we keep as is or update to use the shared branding file instead?

Note: three additional files and six translations - five for `-brand-name-mozilla` and one for `-brand-name-firefox` - would be affected.

Edit: typos. I know it's been quote replied twice now, but correcting for future reference to anyone reading.

**Comment by dschom** [comment]
> r+ from me but I do have a question.
> 
> `-product-mozilla-account` and `-product-mozilla-account` are both declared in `packages/fxa-shared/l10n/branding.ftl` with the former replacing `-brand-name-firefox` in `NewUserEmailForm` and `routes/Checkout`. However, `packages/fxa-shared/l10n/branding.ftl` has `-brand-mozilla` and `-brand-firefox`, yet we're using `-brand-name-mozilla` and `-brand-name-firefox` defined in `packages/fxa-payments-server/src/branding.ftl` instead.
> 
> I'm curious why it's structured this way. Should we keep as is or update to use the shared branding file instead?
> 
> Note: three additional files and six translations - five for `-brand-name-mozilla` and one for `-brand-name-firefox` - would be affected.

Good catch. I'd say we should be using the branding file in fxa-shared... Let me look into this.

**Comment by bcolsson** [comment]
> > r+ from me but I do have a question.
> > `-product-mozilla-account` and `-product-mozilla-account` are both declared in `packages/fxa-shared/l10n/branding.ftl` with the former replacing `-brand-name-firefox` in `NewUserEmailForm` and `routes/Checkout`. However, `packages/fxa-shared/l10n/branding.ftl` has `-brand-mozilla` and `-brand-firefox`, yet we're using `-brand-name-mozilla` and `-brand-name-firefox` defined in `packages/fxa-payments-server/src/branding.ftl` instead.
> > I'm curious why it's structured this way. Should we keep as is or update to use the shared branding file instead?
> > Note: three additional files and six translations - five for `-brand-name-mozilla` and one for `-brand-name-firefox` - would be affected.
> 
> @sardesam Good catch. I'd say we should be using the branding file in [fxa-shared](https://mozilla-hub.atlassian.net/browse/FXA-shared)... Let me look into this.

@sardesam - Good catch. Yes, I'd recommend using brands from `branding.ftl` going forward. Last year some work went into making a shared branding file so for historical reasons, older strings are using different term IDs that only exist in `payments.ftl`.

File: packages/fxa-payments-server/src/components/Header/en.ftl [hunk 3]
```diff
@@ -1,3 +1,4 @@
 ## Component - Header
 
 brand-name-firefox-logo = { -brand-name-firefox } logo
+brand-name-mozilla-logo = { -brand-name-mozilla } logo
```
Comment (bcolsson) [inline packages/fxa-payments-server/src/components/Header/en.ftl:4]:
```suggestion
brand-name-mozilla-logo = { -brand-mozilla } logo
```

**Comment by dschom** [comment]
@bcolsson @sardesam I am looking at the [grunt.js](https://github.com/mozilla/fxa/blob/8ccf65de04956a64f2b9d4386ad08f4edf47a8d0/packages/fxa-payments-server/Gruntfile.js#L7) file in fxa-payments. Sounds like in order to use the branding file in fxa-shared I'd need to alter the grunt file and from that comment it sounds like this could potentially introduce conflicts in l10n strings. 

Because of this, I am thinking we should just stick with what is available in fxa-payments-server. Let me know what you think. If you all think we should go for it, I am happy to do so, but I'd mostly just like to get these changes landed and limit any possible side effects.

**Comment by bcolsson** [comment]
@dschom - it looks like we're already using `{ -brand-mozilla }` in some strings.

This is getting deep into how FxA bundles and uses strings, so I'm not 100% sure I fully understand it, but this [PR](https://github.com/mozilla/fxa/pull/15535) seems like it may have resolved the issue?

**Comment by bcolsson** [comment]
Although trying to test if these terms work in storybook I'm getting an error for the [NewUserEmailForm](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/latest/fxa-payments-server/index.html?path=/story/components-newuseremailform--default):

**Comment by dschom** [comment]
> This is getting deep into how FxA bundles and uses strings, so I'm not 100% sure I fully understand it, but this https://github.com/mozilla/fxa/pull/15535 seems like it may have resolved the issue?

I don't think that's the case. I believe that PR actually **removed** the reference to the shared branding file in order to avoid l10n id conflicts. See diff [here](https://github.com/mozilla/fxa/pull/15535/files#r1336403374).

**Comment by dschom** [comment]
> Although trying to test if these terms work in storybook I'm getting an error for the [NewUserEmailForm](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/latest/fxa-payments-server/index.html?path=/story/components-newuseremailform--default):

@bcolsson This appears to be an existing issue, but I think it'll be easy enough to fix. I'll push a change shortly.

**Comment by sardesam** [comment]
> > Although trying to test if these terms work in storybook I'm getting an error for the [NewUserEmailForm](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/latest/fxa-payments-server/index.html?path=/story/components-newuseremailform--default):
> 
> @bcolsson This appears to be an existing issue, but I think it'll be easy enough to fix. I'll push a change shortly.

As per my chat with @dschom. I'll take care of this. `PaymentProcessing` is also affected.

**Comment by dschom** [comment]
@bcolsson I've updated the strings, so that the reference are in line with the`fxa-shared/l10n/branding.ftl` file. @sardesam is looking into the issue with story book. I'll rebase as soon as here change lands and let you know.

**Comment by sardesam** [comment]
> @bcolsson I've updated the strings, so that the reference are in line with the`fxa-shared/l10n/branding.ftl` file. @sardesam is looking into the issue with story book. I'll rebase as soon as here change lands and let you know.

Storybook issue: https://github.com/mozilla/fxa/pull/15842

**Review by sardesam — APPROVED** [review]
r+

File: packages/fxa-payments-server/src/components/Header/en.ftl [hunk 4]
```diff
@@ -1,3 +1,4 @@
 ## Component - Header
 
 brand-name-firefox-logo = { -brand-name-firefox } logo
+brand-name-mozilla-logo = { -brand-mozilla } logo
```
Comment (bcolsson) [inline packages/fxa-payments-server/src/components/Header/en.ftl:4]: Based on Storybook it appears terms from `shared/branding.ftl` are working with payments as well.

**Review by bcolsson — APPROVED** [review]

**Pull request merged**